packages/bsp/common/etc/update-motd.d/30-armbian-sysinfo

```diff
@@ -156,7 +156,7 @@ display "Battery" "$battery_percent" "20" "1" "%" "$status_battery_text" ; a=$((
 (( $a > 0 )) && echo "" # new line only if some value is displayed
 
 line=0
-if [[ "$(command -v vnstat)" && $(vnstat --iflist | grep -c "$PRIMARY_INTERFACE") -eq 1 && $(vnstat -i $PRIMARY_INTERFACE --oneline | cut -d";" -f4,5) != *Error* ]]; then
+if vnstat -i "$PRIMARY_INTERFACE" &> /dev/null && [[ -n "$PRIMARY_INTERFACE" ]]; then
```

Comment:
it's not necessary to check if PRIMARY_INTERFACE is empty, but if you do want to check that, I suggest doing it _before_  using, not after, i.e:
```
if [[ -n "$PRIMARY_INTERFACE" ]] && vnstat -i "$PRIMARY_INTERFACE" &> /dev/null 
```
But as I mentioned, you don't actually need to check if PRIMARY_INTERFACE is empty, because the command will fail if it is, and you're checking for the command failing.